### ISSUE TITLE
dsbn

### ISSUE THREAD

**Issue by Sir-Death** [wgchang/DSBN#1]
`pred_s, f_s = model(x_s, src_idx * torch.ones(x_s.shape[0], dtype=torch.long).cuda(args.gpu),
                                    with_ft=True)`
`pred_t, f_t = model(x_t, trg_idx * torch.ones(x_t.shape[0], dtype=torch.long).cuda(args.gpu),
                                    with_ft=True)`
I want to know what src_idx * torch.ones(x_s.shape[0], dtype=torch.long) and trg_idx * torch.ones(x_t.shape[0], dtype=torch.long) for

**Comment by JosephPai** [comment]
Same problem. 
Have you figured it out?

**Comment by woozch** [comment]
sorry for the late reply. https://github.com/wgchang/DSBN/blob/8e7e276e846b48b6880578d2984762612d7ec56f/trainval_multi.py#L374
It seems that you mentioned this line of code.
DSBN model can have multiple domain labels.
if the # of the source is N and the # of target is M, there are N+M DSBN in total.
In the code, source domain index(src_idx) should be 0, 1, ... , N-1 and the target domain index (trg_idx) should be N, N+1, ..., N+M-1. (note that trg_idx starts from N)
By multiplying one vector, every sample of x_s and x_t can be labeled as src_idx and trg_idx.
